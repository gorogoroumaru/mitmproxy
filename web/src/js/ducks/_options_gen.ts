/** Auto-generated by test_app.py:TestApp.test_generate_options_js */
export interface OptionsState {
    server: boolean
    showhost: boolean
    add_upstream_certs_to_client_chain: boolean
    confdir: string
    certs: string[]
    cert_passphrase: string | undefined
    ciphers_client: string | undefined
    ciphers_server: string | undefined
    client_certs: string | undefined
    ignore_hosts: string[]
    allow_hosts: string[]
    listen_host: string
    listen_port: number
    mode: string
    upstream_cert: boolean
    http2: boolean
    websocket: boolean
    rawtcp: boolean
    ssl_insecure: boolean
    ssl_verify_upstream_trusted_confdir: string | undefined
    ssl_verify_upstream_trusted_ca: string | undefined
    tcp_hosts: string[]
    content_view_lines_cutoff: number
    key_size: number
    body_size_limit: string | undefined
    keep_host_header: boolean
    block_global: boolean
    block_private: boolean
    block_list: string[]
    anticache: boolean
    anticomp: boolean
    client_replay: string[]
    command_history: boolean
    export_preserve_original_ip: boolean
    onboarding: boolean
    onboarding_host: string
    onboarding_port: number
    proxyauth: string | undefined
    connection_strategy: string
    proxy_debug: boolean
    scripts: string[]
    server_replay_kill_extra: boolean
    server_replay_nopop: boolean
    server_replay_refresh: boolean
    server_replay_use_headers: string[]
    server_replay: string[]
    server_replay_ignore_content: boolean
    server_replay_ignore_params: string[]
    server_replay_ignore_payload_params: string[]
    server_replay_ignore_host: boolean
    server_replay_ignore_port: boolean
    map_remote: string[]
    map_local: string[]
    modify_body: string[]
    modify_headers: string[]
    stickyauth: string | undefined
    stickycookie: string | undefined
    stream_large_bodies: string | undefined
    save_stream_file: string | undefined
    save_stream_filter: string | undefined
    tls_version_client_min: string
    tls_version_client_max: string
    tls_version_server_min: string
    tls_version_server_max: string
    upstream_auth: string | undefined
    web_open_browser: boolean
    web_debug: boolean
    web_port: number
    web_host: string
    web_columns: string[]
    intercept_active: boolean
    intercept: string | undefined
    rfile: string | undefined
    readfile_filter: string | undefined
    web_static_viewer: string | undefined
    view_filter: string | undefined
    view_order: string
    view_order_reversed: boolean
    console_focus_follow: boolean
    termlog_verbosity: string
}

export type Option = keyof OptionsState

export const defaultState: OptionsState = {
    server: true,
    showhost: false,
    add_upstream_certs_to_client_chain: false,
    confdir: "~/.mitmproxy",
    certs: [],
    cert_passphrase: undefined,
    ciphers_client: undefined,
    ciphers_server: undefined,
    client_certs: undefined,
    ignore_hosts: [],
    allow_hosts: [],
    listen_host: "",
    listen_port: 8080,
    mode: "regular",
    upstream_cert: true,
    http2: true,
    websocket: true,
    rawtcp: true,
    ssl_insecure: false,
    ssl_verify_upstream_trusted_confdir: undefined,
    ssl_verify_upstream_trusted_ca: undefined,
    tcp_hosts: [],
    content_view_lines_cutoff: 512,
    key_size: 2048,
    body_size_limit: undefined,
    keep_host_header: false,
    block_global: true,
    block_private: false,
    block_list: [],
    anticache: false,
    anticomp: false,
    client_replay: [],
    command_history: true,
    export_preserve_original_ip: false,
    onboarding: true,
    onboarding_host: "mitm.it",
    onboarding_port: 80,
    proxyauth: undefined,
    connection_strategy: "eager",
    proxy_debug: false,
    scripts: [],
    server_replay_kill_extra: false,
    server_replay_nopop: false,
    server_replay_refresh: true,
    server_replay_use_headers: [],
    server_replay: [],
    server_replay_ignore_content: false,
    server_replay_ignore_params: [],
    server_replay_ignore_payload_params: [],
    server_replay_ignore_host: false,
    server_replay_ignore_port: false,
    map_remote: [],
    map_local: [],
    modify_body: [],
    modify_headers: [],
    stickyauth: undefined,
    stickycookie: undefined,
    stream_large_bodies: undefined,
    save_stream_file: undefined,
    save_stream_filter: undefined,
    tls_version_client_min: "TLS1_2",
    tls_version_client_max: "UNBOUNDED",
    tls_version_server_min: "TLS1_2",
    tls_version_server_max: "UNBOUNDED",
    upstream_auth: undefined,
    web_open_browser: true,
    web_debug: false,
    web_port: 8081,
    web_host: "127.0.0.1",
    web_columns: ["tls", "icon", "path", "method", "status", "size", "time"],
    intercept_active: false,
    intercept: undefined,
    rfile: undefined,
    readfile_filter: undefined,
    web_static_viewer: "",
    view_filter: undefined,
    view_order: "time",
    view_order_reversed: false,
    console_focus_follow: false,
    termlog_verbosity: "info",
}
