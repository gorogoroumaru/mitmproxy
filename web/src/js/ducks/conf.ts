/**
 * Conf houses properties about the current mitmproxy instance that are not options,
 * e.g. the list of available content views or the current version.
 */
import {Reducer} from "redux";

interface ConfState {
    static: boolean
    version: string
    contentViews: string[]
}

// @ts-ignore
export const defaultState: ConfState = window.MITMWEB_CONF || {
    static: false,
    version: "1.2.3",
    contentViews: ["Auto", "Raw"],
};

const reducer: Reducer<ConfState> = (state = defaultState, action): ConfState => {
    return state
}
export default reducer
