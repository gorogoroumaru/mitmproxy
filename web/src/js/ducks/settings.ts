import {fetchApi} from '../utils'
import {AnyAction, Reducer} from "redux";

export const RECEIVE = 'SETTINGS_RECEIVE'
export const UPDATE = 'SETTINGS_UPDATE'
export const REQUEST_UPDATE = 'REQUEST_UPDATE'

// We currently have everything optional, alternatively we could also provide defaults.
interface SettingsState {
    version?: string
    mode?: string
    intercept_active?: boolean
    intercept?: string
    showhost?: boolean
    upstream_cert?: boolean
    rawtcp?: boolean
    http2?: boolean
    websocket?: boolean
    anticache?: boolean
    anticomp?: boolean
    stickyauth?: string
    stickycookie?: string
    stream?: string
    contentViews?: string[]
    listen_host?: string
    listen_port?: number
    server?: boolean
}

const defaultState: SettingsState = {
}

const reducer: Reducer<SettingsState> = (state = defaultState, action): SettingsState => {
    switch (action.type) {

        case RECEIVE:
            return action.data

        case UPDATE:
            return {
                ...state,
                ...action.data,
            }

        default:
            return state
    }
}
export default reducer

export function update(settings) {
    fetchApi.put('/settings', settings)
    return {type: REQUEST_UPDATE}
}

export function addInterceptFilter(example) {
    return (dispatch, getState) => {
        let intercept = getState().settings.intercept;
        if (intercept && intercept.includes(example)) {
            return
        }
        if (!intercept) {
            intercept = example
        } else {
            intercept = `${intercept} | ${example}`
        }
        dispatch(update({intercept}));
    }
}
