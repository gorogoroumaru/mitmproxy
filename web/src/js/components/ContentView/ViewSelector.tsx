import React from 'react'
import {setContentView} from '../../ducks/ui/flow';
import Dropdown, {MenuItem} from '../common/Dropdown'
import {useAppDispatch, useAppSelector} from "../../ducks";


export default React.memo(function ViewSelector() {
    const dispatch = useAppDispatch(),
        contentViews = useAppSelector(state => state.conf.contentViews || []),
        activeView = useAppSelector(state => state.ui.flow.contentView);

    let inner = <span><b>View:</b> {activeView.toLowerCase()} <span className="caret"/></span>

    return (
        <Dropdown
            text={inner}
            className="btn btn-default btn-xs pull-left"
            options={{placement: "top-start"}}>
            {contentViews.map(name =>
                <MenuItem key={name} onClick={() => dispatch(setContentView(name))}>
                    {name.toLowerCase().replace('_', ' ')}
                </MenuItem>
            )}
        </Dropdown>
    )
});
