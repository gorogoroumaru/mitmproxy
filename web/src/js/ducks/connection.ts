import {Reducer} from "redux";
import {OptionsState} from "./_options_gen";

export enum ConnectionState {
    INIT= "CONNECTION_INIT",
    FETCHING = "CONNECTION_FETCHING", // WebSocket is established, but still fetching resources.
    ESTABLISHED = "CONNECTION_ESTABLISHED",
    ERROR = "CONNECTION_ERROR",
    OFFLINE = "CONNECTION_OFFLINE", // indicates that there is no live (websocket) backend.
}


interface ConnState {
    state: ConnectionState,
    message?: string
}

const defaultState: ConnState = {
    state: ConnectionState.INIT,
    message: undefined,
}

const reducer: Reducer<ConnState> = (state = defaultState, action) => {
    switch (action.type) {
        case ConnectionState.ESTABLISHED:
        case ConnectionState.FETCHING:
        case ConnectionState.ERROR:
        case ConnectionState.OFFLINE:
            return {
                state: action.type,
                message: action.message
            }

        default:
            return state
    }
}
export default reducer

export function startFetching() {
    return { type: ConnectionState.FETCHING }
}

export function connectionEstablished() {
    return { type: ConnectionState.ESTABLISHED }
}

export function connectionError(message) {
    return { type: ConnectionState.ERROR, message }
}
export function setOffline() {
    return { type: ConnectionState.OFFLINE }
}
