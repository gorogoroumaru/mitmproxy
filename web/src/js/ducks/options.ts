import {fetchApi} from "../utils"
import * as optionsEditorActions from "./ui/optionsEditor"
import _ from "lodash"
import {Reducer} from "redux";
import {defaultState, Option, OptionsState} from "./_options_gen";
import {AppThunk} from "./index";

export const RECEIVE = 'OPTIONS_RECEIVE'
export const UPDATE = 'OPTIONS_UPDATE'
export const REQUEST_UPDATE = 'REQUEST_UPDATE'

export {Option, defaultState}

const reducer: Reducer<OptionsState> = (state = defaultState, action) => {

    switch (action.type) {
        case RECEIVE:
            let s = <OptionsState>{};
            // @ts-ignore
            for (const [name, value] of Object.entries(action.data)) {
                s[name] = value
            }
            return s;

        case UPDATE:
            let s2 = {...state}
            // @ts-ignore
            for (const [name, value] of Object.entries(action.data)) {
                s2[name] = value
            }
            return s2

        default:
            return state
    }
}
export default reducer

export function pureSendUpdate(option: Option, value, dispatch) {
    return async dispatch => {
        try {
            const response = await fetchApi.put('/options', {[option]: value});
            if (response.status === 200) {
                dispatch(optionsEditorActions.updateSuccess(option))
            } else {
                throw await response.text()
            }
        } catch (error) {
            return dispatch(optionsEditorActions.updateError(option, error))
        }
    }
}

let sendUpdate = _.throttle(pureSendUpdate, 500, {leading: true, trailing: true})

export function update(name: Option, value: any): AppThunk {
    return dispatch => {
        dispatch(optionsEditorActions.startUpdate(name, value))
        sendUpdate(name, value, dispatch);
    }
}

export function save() {
    return dispatch => fetchApi('/options/save', {method: 'POST'})
}

export function addInterceptFilter(example) {
    return (dispatch, getState) => {
        let intercept = getState().options.intercept;
        if (intercept && intercept.includes(example)) {
            return
        }
        if (!intercept) {
            intercept = example
        } else {
            intercept = `${intercept} | ${example}`
        }
        dispatch(update("intercept", intercept));
    }
}
