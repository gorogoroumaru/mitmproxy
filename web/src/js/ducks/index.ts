import {applyMiddleware, combineReducers, compose, createStore} from "redux"
import eventLog from "./eventLog"
import flows from "./flows"
import settings from "./settings"
import ui from "./ui/index"
import connection from "./connection"
import options from './options'
import thunk from "redux-thunk";
import {logger} from 'redux-logger'
import {TypedUseSelectorHook, useDispatch, useSelector} from "react-redux";


const middlewares = [thunk];

// logger must be last
if (process.env.NODE_ENV !== 'production') {
    middlewares.push(logger);
}

// @ts-ignore
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;


export const rootReducer = combineReducers({
    eventLog,
    flows,
    settings,
    connection,
    ui,
    options,
});

export const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(...middlewares))
)
export type RootState = ReturnType<typeof rootReducer>
export type AppDispatch = typeof store.dispatch

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
