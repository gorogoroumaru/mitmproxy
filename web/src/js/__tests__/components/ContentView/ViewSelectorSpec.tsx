import React from 'react'
import renderer from 'react-test-renderer'
import ViewSelector from '../../../components/ContentView/ViewSelector'
import { Provider } from 'react-redux'
import { TStore } from '../../ducks/tutils'


describe('ViewSelector Component', () => {
    let store = TStore(),
        viewSelector = renderer.create(
            <Provider store={store}>
                <ViewSelector/>
            </Provider>
        ),
        tree = viewSelector.toJSON()

    it('should render correctly', () => {
        expect(tree).toMatchSnapshot()
    })
})
