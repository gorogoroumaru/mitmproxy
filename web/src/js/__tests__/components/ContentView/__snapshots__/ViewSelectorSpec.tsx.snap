// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSelector Component should render correctly 1`] = `
<a
  className="btn btn-default btn-xs pull-left"
  href="#"
  onClick={[Function]}
>
  <span>
    <b>
      View:
    </b>
     
    auto
     
    <span
      className="caret"
    />
  </span>
</a>
`;
